examples/ex33.hpp

```diff
@@ -0,0 +1,394 @@
+//              Implementation of the AAA algorithm
+//
+//   Here, we implement the triple-A algorithm [1] for the rational approximation
+//   of complex-valued functions,
+//
+//          p(z)/q(z) ≈ f(z).
+//
+//   In this file, we always assume f(z) = z^{-α}. The triple-A algorithm
+//   provides a robust, accurate approximation in rational barycentric form.
+//   This representation must be transformed into a partial fraction
+//   representation in order to be used to solve a spectral FPDE.
+//
+//   More specifically, we first expand the numerator in terms of the zeros of
+//   the rational approximation,
+//
+//          p(z) ∝ Π_i (z - z_i),
+//
+//   and expand the denominator in terms of the poles of the rational
+//   approximation,
+//
+//          q(z) ∝ Π_i (z - p_i).
+//
+//   We then use these zeros and poles to derive the partial fraction
+//   expansion
+//
+//          f(z) ≈ p(z)/q(z) = Σ_i c_i / (z - p_i).
+//
+//
+//   REFERENCE
+//
+//   [1] Nakatsukasa, Y., Sète, O., & Trefethen, L. N. (2018). The AAA
+//       algorithm for rational approximation. SIAM Journal on Scientific
+//       Computing, 40(3), A1494-A1522.
+
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+/**
+ * RationalApproximation_AAA : compute the rational approximation (RA) of
+ * data @a val at the set of points @a pt.
+ *
+ *  INPUT:  @a val       = vector of data values
```

Comment:
Standard doxygen practice for parameters (see [this](https://github.com/mfem/mfem/blob/fe7f097e699d83aa4d545d53a6bba64b80d91146/linalg/ode.hpp#L73) for an example) will improve the resulting doxygen.

Reply:
@brendankeith , can you address ☝️

Reply:
Thanks for pointing that out, @jamiebramwell !